client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx

```diff
@@ -33,12 +35,23 @@ const GoalsStep: Step = ( { navigation } ) => {
 	const subHeaderText = translate( 'Tell us what would you like to accomplish with your website.' );
 
 	const goals = useSelect( ( select ) => select( ONBOARD_STORE ).getGoals() );
-	const intent = useSelect( ( select ) => select( ONBOARD_STORE ).getIntent() );
-	const { setGoals } = useDispatch( ONBOARD_STORE );
+	const { setGoals, setIntent, clearImportGoal } = useDispatch( ONBOARD_STORE );
 
-	const recordGoalsSelectTracksEvent = () => {
-		const { serializeGoals } = Onboard.utils;
+	useEffect( () => {
+		clearImportGoal();
+	}, [ clearImportGoal ] );
 
+	const handleChange = ( goals: Onboard.SiteGoal[] ) => {
+		const intent = goalsToIntent( goals );
+		setIntent( intent );
+		setGoals( goals );
+		return intent;
+	};
+
+	const recordGoalsSelectTracksEvent = (
```

Comment:
**Suggestion for the future.** Both record functions don't have dependencies with the outer context, depending only on the parameters. It seems to be a good opportunity to encapsulate record tracks events in a separate hook, something like useRecordTracksEvents, returning the two callback functions in an object.

That way the `GoalsStep` would be responsible for rendering the step itself and handling submission, just calling the record functions at the right time. Do you have any thoughts on that?